presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloMetadata.java

```diff
@@ -0,0 +1,548 @@
+        }
+        return bldr.build();
+    }
+
+    /**
+     * Gets all views in the given schema, or all schemas if null
+     *
+     * @param session Current client session
+     * @param schemaNameOrNull Schema to list for the views, or null to list all schemas
+     * @return List of views
+     */
+    @Override
+    public List<SchemaTableName> listViews(ConnectorSession session, String schemaNameOrNull)
+    {
+        return listViews(schemaNameOrNull);
+    }
+
+    /**
+     * Gets all views in the given schema, or all schemas if null
+     *
+     * @param schemaNameOrNull Schema to list for the views, or null to list all schemas
+     * @return List of views
+     */
+    private List<SchemaTableName> listViews(String schemaNameOrNull)
+    {
+        ImmutableList.Builder<SchemaTableName> bldr = ImmutableList.builder();
+
+        if (schemaNameOrNull == null) {
+            for (String schema : client.getSchemaNames()) {
+                for (String view : client.getViewNames(schema)) {
+                    bldr.add(new SchemaTableName(schema, view));
+                }
+            }
+        }
+        else {
+            for (String view : client.getViewNames(schemaNameOrNull)) {
+                bldr.add(new SchemaTableName(schemaNameOrNull, view));
+            }
+        }
+
+        return bldr.build();
+    }
+
+    /**
+     * Begin an insert of data into an Accumulo table. This is for new inserts, not for a CTAS.
+     *
+     * @param session Current client session
+     * @param tableHandle Table handle for the insert
+     * @return Insert handle for the table
+     */
+    @Override
+    public ConnectorInsertTableHandle beginInsert(ConnectorSession session,
+            ConnectorTableHandle tableHandle)
+    {
+        // This is all metadata management for the insert op
+        // Not much to do here but validate the type and return the new table handle (which is the
+        // same)
+        return checkType(tableHandle, AccumuloTableHandle.class, "table");
+    }
+
+    /**
+     * Commit the insert
+     *
+     * @param session Current client session
+     * @param insertHandle Table handle
+     * @param fragments Collection of fragment metadata
+     */
+    @Override
+    public void commitInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle,
+            Collection<Slice> fragments)
+    {
+        // Seems like most connectors just return metadata about the fragments
+        // Unsure if we can use this for an optimization?
+        // Priority is on reading though, so we can investigate this later
+        checkType(insertHandle, AccumuloTableHandle.class, "table");
+    }
+
+    /**
+     * Rollback the insert, which is not supported by this connector
+     *
+     * @param session Current client session
+     * @param insertHandle Table handle
+     */
+    @Override
+    public void rollbackInsert(ConnectorSession session, ConnectorInsertTableHandle insertHandle)
```

Comment:
Don't implement this method if you have nothing to do here. Leave default implementation from base class (which is currently no-op).